src/ui/public/state_management/state.js

```diff
@@ -230,23 +221,28 @@ export default function StateProvider(Private, $rootScope, $location, config) {
       return rison.encode(state);
     }
 
-    try {
-      const hash = this._hashingStore.hashAndSetItem(state);
+    // We need to strip out Angular-specific properties.
+    const json = angular.toJson(state);
+    const hash = createStateHash(json);
+    const isItemSet = hashedItemStore.setItem(hash, json);
```

Comment:
what if we created a `shortenHash(hashedItemStore, hash, json)` function here that brought back the short hashes? I think they are much better looking and the [logic is pretty simple](https://github.com/spalger/kibana/blob/8d1ba1776680d66c65c64350522e48f2b5319222/src/ui/public/state_management/state_storage/hashing_store.js#L79-L90) in my opinion.